packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/marker_test.dart

```diff
@@ -86,6 +86,7 @@ void main() {
         'rotation': 100.0,
         'visible': false,
         'zIndex': 100.0,
+        'zIndexInt': 100,
       });
     });
```

Comment:
Per the comment on the main PR, this file needs tests that various combinations of passing zIndex and zIndexInt work as expected.